io/disk/arcconf/arcconf_raid_oper.py

```diff
@@ -0,0 +1,252 @@
+            disk_val += " %s %s" % (self.channel_no, val)
+            loop_count += 1
+            # Format all the physical disk drives
+            cmd = "echo y | arcconf task start %s device %s %s \
+                  INITIALIZE" % (self.crtl_no, self.channel_no, val)
+            self.check_pass(cmd, "Failed to Format drive")
+
+            # Raid create
+            if loop_count > 1:
+                self.raid_create(disk_val, loop_count, type_name)
+
+    def raid_create(self, disk_data, cnt, type_name1):
+        """
+        function which decides on RAID level
+        TODO 50, 60 and 5EE
+        """
+        self.log.info(" Raid creation with Drives %s, option %s"
+                      % (cnt, type_name1))
+        if cnt == 2:
+            raid = "0"
+            self.raid_exec(type_name1, disk_data, cnt, raid)
+            time.sleep(10)
+
+            raid = "1"
+            self.raid_exec(type_name1, disk_data, cnt, raid)
+            time.sleep(10)
+
+        elif cnt == 3:
+            raid = "1E"
+            self.raid_exec(type_name1, disk_data, cnt, raid)
+            time.sleep(10)
+
+        elif cnt == 4:
+            raid = "10"
+            self.raid_exec(type_name1, disk_data, cnt, raid)
+            time.sleep(10)
+
+            raid = "5"
+            self.raid_exec(type_name1, disk_data, cnt, raid)
+            time.sleep(10)
+
+            raid = "6"
+            self.raid_exec(type_name1, disk_data, cnt, raid)
+            time.sleep(10)
+
+    def raid_exec(self, name1, dsk_data, cnt, raid):
+        """
+        function to create different raid functions
+        """
+        self.log.info(" Creating RAID %s with drives %s"
+                      % (raid, dsk_data))
+        cmd = "echo y | arcconf create %s LOGICALDRIVE %s MAX %s %s" \
+            % (self.crtl_no, name1, raid, dsk_data)
+        self.check_pass(cmd, "Failed to create RAID %s" % raid)
+
+        if raid == 0:
+            self.format_logical(cnt)
+        self.make_fs()
+        time.sleep(10)
+
+        self.log.info(" Deleting RAID %s with drives %s"
+                      % (raid, dsk_data))
+        cmd = "echo y | arcconf delete %s logicaldrive %s" % \
+            (self.crtl_no, self.logicaldrive)
+        self.check_pass(cmd, "Failed to delete RAID %s" % raid)
+
+    def format_logical(self, count1):
+        """
+        format logical drive
+        """
+        self.log.info("Formatting Logical drive %s, having %s drives"
+                      % (self.logicaldrive, count1))
+        cmd = "echo y | arcconf task start %s LOGICALDRIVE %s CLEAR" \
+            % (self.crtl_no, self.logicaldrive)
+        self.check_pass(cmd, "Failed to format Logical Drive")
+        cmd = "arcconf getconfig %s LD" % self.crtl_no
+        self.check_pass(cmd, "Failed to display Logical device details")
+
+    def make_fs(self):
+        """
+        creates filesystem
+        """
+        cmd = "lsscsi  | grep LogicalDrv | awk \'{print $7}\'"
```

Comment:
Looked at the code lv_utils.  filesystem creation is part of lv_utils.lv_mount function,   which accepts vg_name and lv_name as parameters

since we are creating filesystem on non lv's those parameters wouldn't be applicable. i would retain the existing function as is in the code until a Library is available for non LVs.